### PR TITLE
Quickswap deployment

### PR DESCRIPTION
**Pull request by AlexSaplin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,9 @@
...
+    "deploy:polygon:quickswap": "forge script src/script/quickswap/QuickswapPolygonDeployment.s.sol:QuickswapPolygonDeployment --rpc-url $(grep MAINNET_RPC .env | cut -d '=' -f2) --private-key $(grep PRIVATE_KEY .env | cut -d '=' -f2) --broadcast --verify --etherscan-api-key $(grep POLYGON_ETHERSCAN_API_KEY .env | cut -d '=' -f2) -vvvv",
     "deploy:rinkeby": "forge script src/script/RinkebyDeployment.s.sol:RinkebyDeployment --rpc-url $(grep RINKEBY_RPC .env | cut -d '=' -f2) --private-key $(grep PRIVATE_KEY .env | cut -d '=' -f2) --broadcast --verify --etherscan-api-key $(grep ETHERSCAN_API_KEY .env | cut -d '=' -f2) -vvvv",
     "deploy:polygon": "forge script src/script/PolygonDeployment.s.sol:RinkebyDeployment --rpc-url $(grep POLYGON_RPC .env | cut -d '=' -f2) --private-key $(grep PRIVATE_KEY .env | cut -d '=' -f2) --broadcast --verify --etherscan-api-key $(grep ETHERSCAN_API_KEY .env | cut -d '=' -f2) -vvvv"
```
Comment (k1rill-fedoseev) [inline package.json:26]: Let's remove these ones as well

File: src/script/ConfigContract.sol [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import "forge-std/Script.sol";
+
+contract ConfigContract is Script {
+    using stdJson for string;
+
+    string chain;
+    string amm;
+
+    struct BaseParams {
+        address wbtc;
```
Comment (k1rill-fedoseev) [inline src/script/ConfigContract.sol:13]: Sir, alphabetically ordered pls - https://github.com/foundry-rs/forge-std/blob/aea0b2685bebc883c09f5554d7fb481e85d0564d/src/Vm.sol#L167-L186

File: src/script/ConfigContract.sol [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    BaseParams public baseParams;
+    AmmParams public ammParams;
+    GovernanceParams public governanceParams;
+
+    function _parseConfigs() internal {
+        string memory root = vm.projectRoot();
+        string memory baseParamsPath = string.concat(root, "/src/script/configs/", chain, "/base.json");
+        string memory ammParamsPath = string.concat(root, "/src/script/configs/", chain, "/", amm, ".json");
+        string memory governanceParamsPath = string.concat(root, "/src/script/configs/", chain, "/", amm, ".json");
```
Comment (k1rill-fedoseev) [inline src/script/ConfigContract.sol:51]: Incorrect path

File: package.json [hunk 4]
```diff
@@ -19,7 +19,9 @@
     "test-v:better": "forge test -vvvv --etherscan-api-key $(grep ETHERSCAN_API_KEY .env | cut -d '=' -f2)",
     "size": "forge build --sizes --force --via-ir",
     "compile": "forge build",
-    "deploy:mainnet": "forge script src/script/MainnetDeployment.s.sol:MainnetDeployment --rpc-url $(grep MAINNET_RPC .env | cut -d '=' -f2) --private-key $(grep PRIVATE_KEY .env | cut -d '=' -f2) --broadcast --verify --etherscan-api-key $(grep ETHERSCAN_API_KEY .env | cut -d '=' -f2) -vvvv",
+    "deploy:mainnet:uniswap": "forge script src/script/uniswap/UniswapMainnetDeployment.s.sol:UniswapMainnetDeployment --rpc-url $(grep MAINNET_RPC .env | cut -d '=' -f2) --private-key $(grep PRIVATE_KEY .env | cut -d '=' -f2) --broadcast --verify --etherscan-api-key $(grep MAINNET_ETHERSCAN_API_KEY .env | cut -d '=' -f2) -vvvv",
```
Comment (k1rill-fedoseev) [inline package.json:22]: This doesn't allow to run a foundry simulation, I think the following would be better:
```shell
yarn deploy:mainnet:uniswap
yarn deploy:mainnet:uniswap --broadcast
yarn deploy:mainnet:uniswap --broadcast --verify
yarn deploy:mainnet:uniswap --broadcast --verify --verifier sourcify
```
So, let's remove `--broadcast --verify` from here.

File: src/script/AbstractDeployment.sol [hunk 5]
```diff
@@ -1,113 +1,71 @@
...
             type(uint256).max
         );
         EIP1967Proxy vaultProxy = new EIP1967Proxy(msg.sender, address(vault), initData);
         vault = Vault(address(vaultProxy));
 
-        setupGovernance(ICDP(address(vault)), factory);
+        setupGovernance(ICDP(address(vault)));
 
         console2.log("Vault", address(vault));
+        vm.serializeAddress(deploymentJson, "Vault", address(vault));
```
Comment (k1rill-fedoseev) [inline src/script/AbstractDeployment.sol:68]: I think we can add a shortcut helper for such lines

File: src/script/ConfigContract.sol [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        baseParams = abi.decode(vm.parseJson(vm.readFile(baseParamsPath)), (BaseParams));
+        ammParams = abi.decode(vm.parseJson(vm.readFile(ammParamsPath)), (AmmParams));
+        governanceParams = abi.decode(vm.parseJson(vm.readFile(governanceParamsPath)), (GovernanceParams));
```
Comment (k1rill-fedoseev) [inline src/script/ConfigContract.sol:54]: Will this revert if the number of fields in JSON and struct are different?

**Review by k1rill-fedoseev — CHANGES_REQUESTED** [review]

File: src/script/ConfigContract.sol (hunk 6 above)
Comment (AlexSaplin) [reply src/script/ConfigContract.sol]: Yes

**Review by k1rill-fedoseev — APPROVED** [review]

**Pull request merged**